transformer_lens/HookedTransformer.py

```diff
@@ -163,6 +163,11 @@ def __init__(
         # Needed for HookPoints to work
         self.setup()
 
+        # Prepend the BOS token to the input as default when the input is a string.
+        # Even for models not explicitly trained with this, heads often use the first position as a resting position
+        # and accordingly lose information from the first token, so this empirically seems to give better results.
+        self.set_prepend_bos(True)
```

Comment:
Shouldn't this use whatever is in `self.cfg.prepend_bos`?

Reply:
Hi Neel, thank you for the review! I don't think that we have something like `cfg.prepend_bos`. Do you mean that we should have `default_prepend_bos` in the config rather than the model? In other words, that it would be better to add `default_prepend_bos` to `HookedTransformerConfig`?

Reply:
I added `default_prepend_bos` to `HookedTransformerConfig`,
```
default_prepend_bos: bool = True
```

and updated `HookedTransformer` to use the value in the config to set `self.prepend_bos`.
```
self.set_default_prepend_bos(self.cfg.default_prepend_bos)
```

Reply:
FWIW, some design considerations: I wanted to use `model.prepend_bos` to control the default behavior (not `model.cfg.default_prepend_bos`) because the values in `HookedTransformerConfig` are intended to be access-only. In fact, `model.cfg.default_prepend_bos` is like a default value for default prepend bos (default of default). I used `model.prepend_bos` instead of `model.default_prepend_bos` because I think it makes it more explicit and clear that we are overriding the value of `self.prepend_bos` when we pass `prepend_bos` as the method parameter.

Reply:
Yeah, I see where you're coming from. I guess I don't see HookedTransformerConfig as access only? There's stuff that was set when the model was pre-trained (eg num layers), but also stuff a user can set (eg what precision to use, how man devices, etc). Possibly it woudl be better to have "fixed, pre-trained model config" and "user set config" class, idk.

Reply:
Oh I see, by 'access-only', I meant something like 'once the values of the config are initialised, they are not updated throughout the lifetime of the process'. For example, I guess we don't expect the users to change the precision or number of devices in the middle of running the process. On the other hand, I think the users might want to change the default `prepend_bos` without restarting the process just to do some tests etc.

Also, since `set_default_prepend_bos()` is a setter method of `model`, it feels a bit weird to change the value of `model.cfg.default_prepend_bos` with a setter of `model`.

Reply:
Actually, since we already added `default_prepend_bos` to the config, how about getting rid of the `set_default_prepend_bos()` and let the user to just do `HookedTransformer.from_pretrained(model_name, ..., default_prepend_bos=False)` if they want to turn off the behavior?